pkg/platform/api/headchef/headchef.go

```diff
@@ -132,10 +132,19 @@ func (b *BuildParams) WriteToRequest(req runtime.ClientRequest, reg strfmt.Regis
 }
 
 func (r *Client) reqBuild(buildReq *headchef_models.V1BuildRequest, buildStatus *BuildStatus) {
+	defClient := retryhttp.DefaultClient
+	timeout := defClient.HTTPClient.Timeout
+	if timeout == 0 {
+		timeout = retryhttp.DefaultTimeout
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
```

Comment:
Can we bake all of this into retryhttp please, no need to see this chunk repeated for every call we make. We just need a HTTP client and a ctx, both of which retryhttp can provide.